eng/common/scripts/Helpers/PSModule-Helpers.ps1

```diff
@@ -102,17 +112,26 @@ function installModule([string]$moduleName, [string]$version, $repoUrl) {
     throw "Failed to install module $moduleName with version $version"
   }
 
+  $script:InstalledModules["${moduleName}"] = $modules
+
   # Unregister repository as it can cause overlap issues with `dotnet tool install`
   # commands using the same devops feed
   Unregister-PSRepository -Name $repoUrl | Out-Null
 
   return $modules[0]
 }
 
+function InstallAndImport-ModuleIfNotInstalled([string]$module, [string]$version) {
+  if ($null -eq (moduleIsInstalled $module $version)) {
+    Install-ModuleIfNotInstalled -WhatIf:$false $module $version | Import-Module
+  } elseif (!(Get-Module -Name $module)) {
```

Comment:
What cases do you expect to hit this elseif block? I would expect if it were not installed it would be installed by the first block.

Reply:
`moduleIsInstalled` function checks whether it's installed or not. If not, we install it and import it.

the `elseif` is the case where the module is _installed_, but not _imported_ yet. This short circuits the import-module call in most cases.